pkg/kubelet/pod_workers.go

```diff
@@ -1519,8 +1519,18 @@ func (p *podWorkers) completeWork(podUID types.UID, phaseTransition bool, syncEr
 		// Network is not ready; back off for short period of time and retry as network might be ready soon.
 		p.workQueue.Enqueue(podUID, wait.Jitter(backOffOnTransientErrorPeriod, workerBackOffPeriodJitterFactor))
 	default:
-		// Error occurred during the sync; back off and then retry.
-		p.workQueue.Enqueue(podUID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		// Error occurred during the sync; back off and then retry. If the error includes a backoff expiration,
+		// wait until then instead of the default to avoid adding extra time to the backoff.
+		var backoffAt time.Time
+		var isBackoffErr bool
+		if backoffAt, isBackoffErr = kubecontainer.MinBackoffExpiration(syncErr); !isBackoffErr {
+			backoffAt = p.clock.Now().Add(p.backOffPeriod)
+		}
+		backoff := backoffAt.Sub(p.clock.Now())
+		if backoff < 0 {
+			backoff = 0
```

Comment:
is `0` handled well by `wait.Jitter` and queue processor? Was it zero ever before?

Reply:
0 is used (without jitter) above in the switch, so the queue processor can definitely handle it. The unit tests don't seem to have an issue with the jitter + 0 delay.